core/src/io/anuke/mindustry/world/modules/PowerModule.java

```diff
@@ -8,13 +8,14 @@
 import java.io.IOException;
 
 public class PowerModule extends BlockModule{
-    public float amount;
+    public float buffer;
+    public float productionEfficiency = 1;
```

Comment:
Why is this in `PowerModule` when it's generator-specific?

Reply:
And buffer is only used by consumers. Where should it go then?

Reply:
In your code, the buffer is used by generators as well as consumers and batteries. It's a general value that all power blocks should have.

Production efficiency is a property of specific generators only, so it should go in `GeneratorEntity`